cosi/check/check.go

```diff
@@ -138,10 +137,12 @@ func checkList(list *onet.Roster, descs []string, detail bool) error {
 // (pass an io.File or use an strings.NewReader for strings). It uses
 // the roster el to create the collective signature.
 // In case the signature fails, an error is returned.
-func signStatement(client *service.Client, read io.Reader, el *onet.Roster) (*service.SignatureResponse,
-	error) {
+func signStatement(client *service.Client, read io.Reader, el *onet.Roster) (*service.SignatureResponse, error) {
 	suite := client.Suite().(kyber.HashFactory)
-	msg, _ := hash.Stream(suite.Hash(), read)
+
+	h := suite.Hash()
+	io.Copy(h, read)
```

Comment:
why not use `h.Write` here? Also in `signStatement`?

Reply:
io.Copy is to go from a io.Reader to an io.Writer. h.Write() is to send a []byte into an io.Writer.  io.Copy allocates a 32k buffer and loops on Read/Write to/from that buffer. If I used h.Write I would need to do the same. See io.Copy's internals here: https://golang.org/src/io/io.go#L361